src/rgw/rgw_reshard.cc

```diff
@@ -460,17 +460,34 @@ static int cancel_reshard(rgw::sal::RadosStore* store,
                           const ReshardFaultInjector& fault,
                           const DoutPrefixProvider *dpp)
 {
+  static constexpr auto max_retries = 10;
   // unblock writes to the current index shard objects
-  int ret = set_resharding_status(dpp, static_cast<rgw::sal::RadosStore*>(store), bucket_info,
-                                  cls_rgw_reshard_status::NOT_RESHARDING);
+  int ret = 0;
+  int tries = 0;
+  do {
+    ret = set_resharding_status(dpp, static_cast<rgw::sal::RadosStore*>(store), bucket_info,
+				cls_rgw_reshard_status::NOT_RESHARDING);
```

Comment:
it looks like `set_resharding_status()` is a write to each of the bucket index objects, and those writes are not covered by cls_version which is where the ECANCELED errors would come from

only the bucket metadata should need this retry handling